internal/services/dns/dns_cname_record_resource.go

```diff
@@ -191,6 +194,15 @@ func resourceDnsCNameRecordRead(d *pluginsdk.ResourceData, meta interface{}) err
 			targetResourceId := ""
 			if props.TargetResource != nil && props.TargetResource.Id != nil {
 				targetResourceId = *props.TargetResource.Id
+				if recordTypeID, err := recordsets.ParseRecordTypeIDInsensitively(targetResourceId); err == nil {
+					targetResourceId = recordTypeID.ID()
+				} else if trafficManagerID, err := endpoints.ParseEndpointTypeIDInsensitively(targetResourceId); err == nil {
+					targetResourceId = trafficManagerID.ID()
+				} else if cdnID, err := cdn.EndpointIDInsensitively(targetResourceId); err == nil {
+					targetResourceId = cdnID.ID()
+				} else if frontDoorID, err := frontdoor.FrontendEndpointIDInsensitively(targetResourceId); err == nil {
+					targetResourceId = frontDoorID.ID()
+				}
```

Comment:
We're still silently ignoring the case where the returned ID is none of those resource types. We should error if the resource type isn't any of these.

Reply:
I needed to check internally but you're right @wuxu92, apologies!

I did push a commit to add a comment above this logic linking to a piece of work we're doing on dynamic resource IDs - once that functionality is available we should be able to update this.